File: Easy/3014_Minimum_Number_of_Pushes_to_Type_Word_I/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    int minimumPushes(string word) {
        int ans = 0;

        for (int i = 0; i < word.size(); i++) {
            ans += i / 8 + 1;
        }

        return ans;
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    string word1 = "abcde";
    cout << solution.minimumPushes(word1) << endl;

    // test cases 2
    string word2 = "xycdefghij";
    cout << solution.minimumPushes(word2) << endl;

    return 0;
}

File: Easy/3014_Minimum_Number_of_Pushes_to_Type_Word_I/Solution.go
package main

import "fmt"

// Past the function from leetcode here
func minimumPushes(word string) int {
	ans := 0

	for i := 0; i < len(word); i++ {
		ans += i/8 + 1
	}

	return ans
}

func main() {
	// test cases 1
	word1 := "abcde"
	fmt.Println(minimumPushes(word1))

	// test cases 2
	word2 := "xycdefghij"
	fmt.Println(minimumPushes(word2))

}
